### DISCUSSION TITLE
Samples calculation in Replay Strategy

### DISCUSSION THREAD

**Discussion by AdityaJainTD** [ContinualAI/avalanche#1105] [Q&A] [answered]
Hi,

I have been using Replay Strategy to train a classification model. The number of samples is fixed to 200. And as in Replay Strategy, each time I train, I will have  200/n_classes_encountered_so_far per examples per class in your memory. So this means, if I have 6 classes then the total samples used in 6 experiences would be as follows :-

_Experience 1 - 200//1 -> 200 (200 samples of class 1)
Experience 2 - 200//2 -> 100 (100 samples of classes 1 and 2)
Experience 3 - 200//3 ->  66 (66 samples of classes 1, 2 and 3)
Experience 4 - 200//4  -> 50  (50 samples of classes 1, 2, 3 and 4)
Experience 5 - 200//5  -> 40 (40 samples of classes 1, 2, 3, 4 and 5)
Experience 6 - 200//6  -> 33 (33 samples of classes 1, 2, 3, 4, 5 and 6)_

=> ~(200*6) => 1200 

So can I say a total of 1200 samples would be used for the training purpose?

**Comment by AntonioCarta** [comment, marked as answer]
> So can I say a total of 1200 samples would be used for the training purpose?

Yes, but this is not what most people mean when they talk about memory size. Usually you only care about the maximum number of samples that you keep in memory *at the same time*.

**Reply by AdityaJainTD** [reply]
@AntonioCarta Thanks for the clarification.
